vignettes/workflows.Rmd

```diff
@@ -0,0 +1,292 @@
+
+2. A density plot of turn transition times. A plot like this is expected to look like a normal distribution centered around 0-200ms (Stivers et al. 2009). Deviations from this may signal problems in the dataset, for instance due to imprecise or automated annotation methods.
+
+3. A scatterplot of turn transition (x) by turn duration (y). This combines both distributions and is expected to look like a cloud of datapoints that is thickest in the middle region. Any standout patterns (for instance, turns whose duration is equal to their transition time) are indicative of problems in the segmentation or timing data.
+
+Each of the three plots can also be generated separately:
+```r
+
+plot_density(data, colname="duration", title="Turn durations",xlab="duration (ms)")
+
+plot_density(data, colname="FTO", title="Turn transitions (FTO)",xlab="FTO (ms)")
+
+plot_scatter(data, colname_x="FTO",colname_y="duration",title="Turn transitions and durations",xlab="transition (ms)", ylab="duration (ms)")
+
+```
+
+
+## Workflow 2: Plot conversations
+
+Another key use of `talkr` is to visualize conversational patterns. A first way to do so is `geom_turn()`, a ggplot2-compatible geom that visualizes the timing and duration of turns in a conversation.
+
+We can start by simply visualizing some of the conversations in the dataset. Here we sample the first four and plot the first minute of each. We display them together using `facet_wrap()` by `source`.
+
+```{r geom_turn_demon1}
+library(ggplot2)
+
+# we simplify participant names
+conv <- data |>
+  dplyr::group_by(source) |>
+  dplyr::mutate(participant = as.character(factor(participant, labels=c("A","B"),ordered=T)))
+
+# select first four conversations
+these_sources <- unique(data$source)[1:4]
+
+conv |>
+  dplyr::filter(end < 60000, # select first 60 seconds
+                source %in% these_sources) |> # filter to keep only these conversations
+  ggplot(aes(x = end, y = participant)) +
+  geom_turn(aes(
+    begin = begin,
+    end = end)) +
+  xlab("Time (ms)") +
+  ylab("") +
+  theme_turnPlot() +
+  facet_wrap(~source) # let's facet to show the conversations side by side
+```
+
+More often, we will want to plot a single conversation and explore it in some more detail. Let's zoom in on one of these first four. If we plot it without further tweaking, it is not the most helpful: the conversation is 15 minutes long and it is hard to appreciate its structure when we put it all on a single line.
+
+```r
+
+conv |>
+  dplyr::filter(source == "/dutch2/DVA12S") |>
+  ggplot(aes(x = begin, y = participant)) +
```

Comment:
best to work with `end` here on the x axis, so not to cut off the end of the conversation. The aes sets up the plot, and you provide the dimensions here, in essence.

Reply:
fixed in https://github.com/elpaco-escience/talkr/commit/ded2d8722334aa2ebd483993589a3fc5017e4cd4